### DISCUSSION TITLE
Node failed to establish connection to cluster

### DISCUSSION THREAD

**Discussion by longansv** [gravitational/teleport#39169] [Q&A]
Hi everyone. Please help me. After I use this command to configure the node I want to add to the teleport.
sudo bash -c "$(curl -kfsSL https://longhd98.click/scripts/be485ab87bba370984dd33fc88364ab5/install-node.sh)"
After installing, I received the following error message:
`● teleport.service - Teleport Service
   Loaded: loaded (/usr/lib/systemd/system/teleport.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2024-03-11 11:11:50 +07; 4min 55s ago
 Main PID: 3357 (teleport)
   CGroup: /system.slice/teleport.service
           └─3357 /usr/local/bin/teleport start --config /etc/teleport.yaml --pid-file=/run/teleport.pid

Mar 11 11:15:28 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:15:28+07:00 INFO [AUTH]      Attempting registration via proxy server. auth/register.go:288
Mar 11 11:15:28 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:15:28+07:00 ERRO [PROC:1]    Node failed to establish connection to cluster: Post "https://longhd98.click:443/v1/webapi/host/credentials": tls: failed to verify certificate: x509: certificate signed by unknown authority. pid:3357.1 service/connect.go:91
Mar 11 11:15:48 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:15:48+07:00 INFO [PROC:1]    Joining the cluster with a secure token. pid:3357.1 service/connect.go:417
Mar 11 11:15:48 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:15:48+07:00 INFO [AUTH]      Attempting registration via proxy server. auth/register.go:288
Mar 11 11:15:48 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:15:48+07:00 ERRO [PROC:1]    Instance failed to establish connection to cluster: Post "https://longhd98.click:443/v1/webapi/host/credentials": tls: failed to verify certificate: x509: certificate signed by unknown authority. pid:3357.1 service/connect.go:91
Mar 11 11:15:50 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:15:50+07:00 WARN [UPLOAD:1]  The Instance connector is still not available, process-wide services such as session uploading will not function pid:3357.1 service/service.go:2863
Mar 11 11:16:17 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:16:17+07:00 INFO [PROC:1]    Joining the cluster with a secure token. pid:3357.1 service/connect.go:417
Mar 11 11:16:17 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:16:17+07:00 INFO [AUTH]      Attempting registration via proxy server. auth/register.go:288
Mar 11 11:16:17 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:16:17+07:00 ERRO [PROC:1]    Node failed to establish connection to cluster: Post "https://longhd98.click:443/v1/webapi/host/credentials": tls: failed to verify certificate: x509: certificate signed by unknown authority. pid:3357.1 service/connect.go:91
Mar 11 11:16:20 jumpserver03.vascloud.vnpt.vn teleport[3357]: 2024-03-11T11:16:20+07:00 WARN [UPLOAD:1]  The Instance connector is still not available, process-wide services such as session uploading will not function pid:3357.1 service/service.go:2863
`
![image](https://github.com/gravitational/teleport/assets/122434297/c569dd34-cc2d-4933-9604-8ad3c8d5690c)

**Comment by webvictim** [comment]
I think we solved this one on Slack - you were using a self-signed certificate, but the connecting node did not have it installed in its system trust store (`/usr/share/ca-certificates`)

**Reply by longansv** [reply]
yes, thank you very much
